src/mpid/ch4/netmod/include/netmod_am_fallback_rma.h

```diff
@@ -124,7 +126,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_put(const void *origin_addr,
                                               MPIDI_winattr_t winattr)
 {
     return MPIDIG_mpi_put(origin_addr, origin_count, origin_datatype, target_rank, target_disp,
-                          target_count, target_datatype, win);
+                          target_count, target_datatype, win, DEFAULT_VCI);
 }
```

Comment:
why only a single vci parameter instead of src and dst?

Reply:
The semantics of one-sided programming only need parameters for the origin side vci. The target side vci is implicit and doesn't matter for the semantics as long as it works. In another word, it works better to leave the target side vci selection to configuration than parameters.

We are taking that flexibility for potential optimizations. For now, we are choosing the target vci as the same as the origin vci. But we could also use any specific vci. One application I am thinking of is to have an async thread polling a dedicated vci and have all RMA target that vci. That way we can easily achieve implicit progress for RMA.

Reply:
The RMA active message has to use consistent vci across netmod and shmmod or at synchronization we won't be able to effectively poke progress. Thus, I have modified the code by initializing per-window `am_vci` at ch4-layer, which will be the lowest denominator between netmod and shmmod. Consequently, we vci selection will happen inside the RMA `MPIDIG_` functions. The commit in reference has now been removed.

Reply:
> The RMA active message has to use consistent vci across netmod and shmmod or at synchronization we won't be able to effectively poke progress. Thus, I have modified the code by initializing per-window `am_vci` at ch4-layer, which will be the lowest denominator between netmod and shmmod. Consequently, we vci selection will happen inside the RMA `MPIDIG_` functions. The commit in reference has now been removed.

I don't want to hold this up, but will just note that per-window VCI seems inflexible for PUT/GET communication. There's no reason you can't use multiple VCIs for those types of unordered operations. This can be left as future optimization work.

Reply:
The problem is at synchronization. We need to progress or check for each vci that has operations issued. And doing that contends with all vci threads.

I think we can have two modes, either do strictly per-window vci, or spread the operations to all vcis and always do global synchronization. The former is good for small messages; the latter is good for large messages. For the latter, the per-window vci assignment can be just ignored.

Reply:
I also need note that the `am_vci` is only for active messages. Currently the native path RMA has its own netmod/shm hooks for progress, so the native path can use independent vci schemes.